### PR TITLE
Update newhltjetmet prompt

### PR DESCRIPTION
**Pull request by physicist87**
We updated prompt monitoring codes of HLTJetMET.

Efficiency of Jet MET and Energy Fractions of HLT Jet are updated.

### CODE REVIEW

**Comment by mtosi** [comment]
thanks !

I think I do not see a 2D map, can you please add it ?

**Comment by mtosi** [comment]
then, I do not understand why you cannot make use of the METMonitor code
if there are missing features there, I would suggest to add them there

File: DQMOffline/Trigger/plugins/JetMETMonitor.cc [hunk 1]
```diff
@@ -0,0 +1,1237 @@
...
+     std::vector<reco::PFJet> jets;
+     if ( int(jetHandle->size()) < njets_ ) return;
+     for ( auto const & jet : *jetHandle ) {
+       jets.push_back(jet);
+     }
+     if ( int(jets.size()) < njets_ ) return;
+     // HLT Jet Selection
+     edm::Handle<reco::PFJetCollection> hltjetHandle;
+     iEvent.getByToken( hltpfjetToken_, hltjetHandle );
+     std::vector<reco::PFJet> hlt_jets;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/JetMETMonitor.cc]: I'm confused, why you are labeling them hlt ? aren't they reco ?

**Comment by mtosi** [comment]
could you point to the DQM file you obtained, please ?

File: DQMOffline/Trigger/plugins/JetMETMonitor.cc (hunk 1 above)
Comment (physicist87) [inline DQMOffline/Trigger/plugins/JetMETMonitor.cc]: To make efficiency plots of Jet, we defined numerator as match offilne jet with hltjet. 
So I used hlt jets. 
Denominator : Offline Jet > 20 GeV
Numerator : Offline Jet (matched hltjet , dR < 0.2 ).

**Comment by physicist87** [comment]
1. When I made these code, I thought that someone was updating the METMonitor code. 
So I thought we needed to avoid crash or problem. So I made JetMETMonitor code based on METMonitor code .

2. I didn't make 2D map. As I remember, you suggest delta Eff. (HEP - HEM) for Eta and Phi. 
I'm trying to make these plots.

3. DQMOutpt : /afs/cern.ch/user/s/sha/public/DQMOutput/DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root
This output is not final output. There are still test plots. But I think it's  enough to check out.

File: DQMOffline/Trigger/plugins/JetMETMonitor.cc [hunk 2]
```diff
@@ -0,0 +1,1237 @@
...
+}
+
+void JetMETMonitor::fillDescriptions(edm::ConfigurationDescriptions & descriptions)
+{
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>  ( "FolderName", "HLT/JetMETMonitor" );
+
+  desc.add<edm::InputTag>( "met",      edm::InputTag("pfMet") );
+  desc.add<edm::InputTag>( "pfjets",     edm::InputTag("ak4PFJetsCHS") );
+  desc.add<edm::InputTag>( "hltpfjets",     edm::InputTag("hltAK4PFJets") ); // ak4jet about HLT
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/JetMETMonitor.cc]: where this collection come from ?
I doubt it is available in RAW format which is the standard format processed @tier0 in the PromptReco
I'm really confused :(

or are you planning to make use of this in the HLTMonitor PD ?
it was not defined in the google doc in this way ....

**Comment by mtosi** [comment]
I understand, but I doubt it works
because the HLT jet collection you are pretending to access should not be
available !
you should make use of the HLT objects available as 4-vectors in the
TriggerResults map, in case
right ?

how did you test your code ?

**Comment by mtosi** [comment]
please, update METMonitor if you need to adapt the current version

please, add 2D plot for the HEP17 monitoring and in general any features
related to the detector

I'm going to give a look at the DQM file, thanks

**Comment by mtosi** [comment]
I'm looking at the root file,
I do not find the efficiency w/ variable binning, please add it
while I see many empty plots, for instance plots related to HF for the path HLT_AK8PFJet500
I think you should add a flag in order to switch such kind of plots only when it makes sense
do you agree ?

then, I would add a ratio between the efficiency you get in HEP17 and HEM17

then, the numerator of the distribution vs LS is empty (at least hltjetpTVsLS)
and you do not have the efficiency vs LS
I think you should add a function in order to handle automatically all your plots

HLT_CaloJet500_NoJetID has 2D plot, but the corresponding efficiency is not available !
please make it for all paths, thanks

for path HLT_PFMET170_HBHECleaned, as written before, please make use of the METMonitor code

it is not that clear to me what you learn from plots in JetEnFraction

File: DQMOffline/Trigger/python/JetMETMonitor_cff.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+CaloJet500_NoJetID_Prommonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_CaloJet500_NoJetID_v*")
+
+# HLT_PFMET170_HBHECleaned
+PFMET170_HBHECleaned_Prommonitoring = hltJetMETmonitoring.clone()
+PFMET170_HBHECleaned_Prommonitoring.FolderName = cms.string('HLT/JetMETMonitor/HLT_PFMET170_HBHECleaned/')
+PFMET170_HBHECleaned_Prommonitoring.ptcut = cms.double(170)
+PFMET170_HBHECleaned_Prommonitoring.ispfjettrg = cms.bool(False)
+PFMET170_HBHECleaned_Prommonitoring.iscalojettrg = cms.bool(False)
+PFMET170_HBHECleaned_Prommonitoring.ismettrg = cms.bool(True)
+PFMET170_HBHECleaned_Prommonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_PFMET170_HBHECleaned_v*")
```
Comment (mtosi) [inline DQMOffline/Trigger/python/JetMETMonitor_cff.py]: are you sure it is available ?

Comment (physicist87) [inline DQMOffline/Trigger/python/JetMETMonitor_cff.py]: I'm still fixing our code.

I think these code are not suitable for Prompt  DQM. (Due to I used hlt jet or met information. )

So I think I need to remove this PR and recreate PR.

Do you are with me?

**Comment by mtosi** [comment]
what about triggers

HLT_PFJet140_v8
HLT_PFJet200_v8
HLT_PFJet260_v9
HLT_PFJet320_v9
HLT_PFJet400_v9

HLT_AK8PFJet40_v*
HLT_AK8PFJet60_v4
HLT_AK8PFJet80_v4
HLT_AK8PFJet140_v4
HLT_AK8PFJet200_v4
HLT_AK8PFJet260_v*
HLT_AK8PFJet320_v*
HLT_AK8PFJet400_v*
HLT_AK8PFJet450_v*
HLT_AK8PFJet500_v*

**Comment by mtosi** [comment]
what about 
HLT_PFJetFwd40_v9
HLT_PFJetFwd60_v9
HLT_PFJetFwd80_v8
HLT_PFJetFwd140_v8
HLT_PFJetFwd200_v8
HLT_PFJetFwd260_v9
HLT_PFJetFwd320_v9
HLT_PFJetFwd400_v9
HLT_PFJetFwd450_v9
HLT_PFJetFwd500_v9
HLT_AK8PFJetFwd40_v*
HLT_AK8PFJetFwd60_v4
HLT_AK8PFJetFwd80_v4
HLT_AK8PFJetFwd140_v4
HLT_AK8PFJetFwd200_v4
HLT_AK8PFJetFwd260_v*
HLT_AK8PFJetFwd320_v*
HLT_AK8PFJetFwd400_v*
HLT_AK8PFJetFwd450_v*
HLT_AK8PFJetFwd500_v*

**Comment by physicist87** [comment]
Milos and Arun asked me for (AK4)PFJet450, AK8PFJet500 and CaloJets500 and (PFMET170 and PFMETTypeOne190 paths) . 

So I used these triggers. And you can see other Jet triggers (PFJet40 60 80 .).

But these trigger are used to confirm that our code are working well ( Due to lack of entries , I just add these triggers.)

**Comment by mtosi** [comment]
which lack of entries ?
we are going to use the multirun harvesting
and the minimum stat to have reasonable distributions (once you defined a
reasonable binning, of course)
has to be defined by the JME group

=> I do not really get your comment
  mia

**Comment by physicist87** [comment]
Actually, If you see my code, I required Offline Jet matched with HLTJet. At here I required that HLTJet should be over than trigger pt threshold.  (I know I can't apply this requirement to prompt validation. There are no hlt jet or met. So I'm still fixing it.)

When I develop these code, I only used one root file. So it is not enough to collect high pt jets. At that time, for trigger path having high pt trigger , many plots are empty. So I just use low trigger path to check my codes.

When I use full mc sample, there is no problem about the lack of entries. But It took much time. So I used one root file to debug my code. 

I'm sorry for getting you confused.

**Comment by mtosi** [comment]
please , try to converge w/ this PR soon
otherwise your trigger paths will not be integrated in the HLT menu !
  mia

**Comment by mtosi** [comment]
can we close this PR ?
there is this one, now
https://github.com/cms-sw/cmssw/pull/19076

**Comment by dmitrijus** [comment]
-1

**Pull request closed without merging**